dashboard/src/modules/components/ProfileComponent/NewKeyModal.jsx

```diff
@@ -21,7 +14,12 @@ const NewKeyModal = (props) => {
   const { isModalOpen, newKeyLabel } = useSelector(
     (state) => state.keyManagement
   );
-
+  const handleKeyPress = (e) => {
+    const key = e.key;
+    if (key === "Enter" && newKeyLabel) {
+      dispatch(sendNewKeyRequest(newKeyLabel));
```

Comment:
Since the expression, `dispatch(sendNewKeyRequest(newKeyLabel))`, now appears twice in this code (lines 20 and 35) and we want it to do the exact same thing in both cases, consider moving it to a helper function and calling the function in these places.

Reply:
Gotcha!